### PR TITLE
fix: Remove ClassVar type declarations for instance variables

### PR DESCRIPTION
**Pull request by kevinjacobs-delfi**
Remove extraneous `ClassVar` declarations for several instance attributes.

### CODE REVIEW

**Comment by kevinjacobs-delfi** [comment]
Yes.  I believe all of your requests have been fulfilled except opening an issue.  You're welcome to ignore my patches, if that is indeed a barrier to contribution.

**Comment by benbenbang** [comment]
Hi @kevinjacobs-delfi,

Thanks for your contributions. While I greatly appreciate your patches, I do ask that all contributors follow the [Contribution Guidelines](https://github.com/benbenbang/types-confluent-kafka/blob/main/.github/CONTRIBUTING.md). Creating an issue before submitting a PR helps everyone track changes more effectively and discuss potential impacts.

If you find this process a barrier, I’d be more than happy to discuss how I can make it easier for you.
Again, I value your input and would love to see your contributions integrated.

**Comment by kevinjacobs-delfi** [comment]
I'm also glad to contribute and am grateful that you and others have provided this package.  

My reservation about opening issues is that I'm adding `confluent_kafka` typing to a fairly large codebase and there are likely to be a range of fixes and tweaks needed.  Most (like this one) are obvious and writing a detailed justification for each is onerous and not a great use of my time.  I'm extremely happy to discuss any suggested changes that are non-obvious or controversial, but I expect most will be fairly pedestrian (like the ones in this PR).

As a compromise, can I open an omnibus issue for a series of PRs rather than one for each PR?  Should any PR be deemed to contain "non-obvious or controversial" suggestions, then we can open an individual issue to capture that discussion?

**Comment by benbenbang** [comment]
Hey,

Thanks for your feedback. I totally agree that grouping similar types of issues makes sense. After reviewing your PR, I also checked the original implementation in C and other modules where I used ClassVar. It looks like I may have made some additional errors, as it was initially a one-week project.

If the changes you’re proposing are all related to `ClassVar` then linking all relevant PRs back to that one makes a lot of sense. This not only helps in tracking but also provides us a great opportunity to correct any mistakes I've made.

**Comment by kevinjacobs-delfi** [comment]
I'm happy to audit all of the other uses of `ClassVar` and add any corrections needed to this PR -- it just happens those were the first that I noticed.  

I have some pending observations around `Message.event()` and `Message.key()` return type annotations that need a bit more exploration, but I'll save those for the next PR.

**Comment by benbenbang** [comment]
Thank you. Feel free to assign some of them to me, as I am also planning to do some fixes. We can also group all the discussions in that issue.

**Comment by benbenbang** [comment]
Hello, thanks for the big effort. There's definitely a big misunderstanding.

What I meant was that we create one issue, and then use multiple PRs link back to an issue, so that we can **split** the work. Because I also started to fix some issues...

**Comment by benbenbang** [comment]
@kevinjacobs-delfi 

Sorry, I might need to stop you here, and we regroup again to do the communication. If you need some help,  I can help you split this PR into several. Also, I will be able to combine my changes together if needed.

**Comment by kevinjacobs-delfi** [comment]
I don't see any value in splitting this PR, but you're welcome to do whatever you wish with it.  I suggest that you merge your changes and I'm happy to take care of reconciling the merge conflicts.

Re misunderstanding:  Not really-- I fixed the `ClassVar` annotations in about 30 minutes.  I spent much more time working on verifying the admin client APIs with incompatible method overloads, which will be submitted after these changes land.

File: confluent_kafka-stubs/cimpl.pyi [hunk 1]
```diff
@@ -430,7 +430,8 @@ class Producer:
         topic: str,
         value: str | bytes | None = None,
         key: str | bytes | None = None,
-        partition: int | None = None,
+        partition: int = -1,
+        callback: Any = None,
```
Comment (kevinjacobs-delfi) [inline confluent_kafka-stubs/cimpl.pyi:434]: Note type change on `partition` and addition of missing `callback` argument (type TBD).

File: confluent_kafka-stubs/schema_registry/schema_registry_client.pyi [hunk 2]
```diff
@@ -68,25 +68,25 @@ class SchemaRegistryClient:
...
 class Schema:
-    schema_str: ClassVar[str]
-    schema_type: ClassVar[str]
-    references: ClassVar[list[SchemaReference]]
+    schema_str: str
+    schema_type: str
+    references: list[SchemaReference]
 
-    def __init__(self, schema_str: str, schema_type: str, references: list[SchemaReference] | None = None) -> None: ...
+    def __init__(self, schema_str: str, schema_type: str, references: list[SchemaReference] = []) -> None: ...
```
Comment (kevinjacobs-delfi) [inline confluent_kafka-stubs/schema_registry/schema_registry_client.pyi:75]: Bad practice to use mutable containers as defaults, but this is how the actual code is defined.

File: confluent_kafka-stubs/serialization/__init__.pyi [hunk 3]
```diff
@@ -10,16 +10,16 @@ from typing import ClassVar
...
+    KEY: ClassVar[str]
+    VALUE: ClassVar[str]
 
 class SerializationContext:
-    topic: ClassVar[str]
-    field: ClassVar[str]
-    headers: ClassVar[str]
+    topic: str
+    field: str
+    headers: list[tuple[str, ...]]
```
Comment (kevinjacobs-delfi) [inline confluent_kafka-stubs/serialization/__init__.pyi:20]: corrected type

File: confluent_kafka-stubs/admin/_config.pyi [hunk 4]
```diff
@@ -29,15 +29,15 @@ class ConfigSource(Enum):
...
-    synonyms: ClassVar[ConfigEntry]
-    incremental_operation: ClassVar[AlterConfigOpType]
+    name: str
+    value: str
+    source: ConfigSource
+    is_read_only: bool
+    is_default: bool
+    is_sensitive: bool
+    is_synonym: bool
+    synonyms: list[ConfigEntry]
```
Comment (kevinjacobs-delfi) [inline confluent_kafka-stubs/admin/_config.pyi:39]: corrected type

File: confluent_kafka-stubs/admin/_config.pyi [hunk 5]
```diff
@@ -29,15 +29,15 @@ class ConfigSource(Enum):
...
+    incremental_operation: AlterConfigOpType | None
```
Comment (kevinjacobs-delfi) [inline confluent_kafka-stubs/admin/_config.pyi:40]: corrected type

File: confluent_kafka-stubs/admin/_config.pyi [hunk 6]
```diff
@@ -48,20 +48,20 @@ class ConfigEntry:
         is_default: bool = False,
         is_sensitive: bool = False,
         is_synonym: bool = False,
-        synonyms: list[ConfigEntry] | None = None,
+        synonyms: list[ConfigEntry] = [],
```
Comment (kevinjacobs-delfi) [inline confluent_kafka-stubs/admin/_config.pyi:51]: bad practice to use a mutable container as a default, but this reflect the type of the actual code

File: confluent_kafka-stubs/admin/_config.pyi [hunk 7]
```diff
@@ -48,20 +48,20 @@ class ConfigEntry:
...
-    set_config_dict: ClassVar[dict[str, str]]
-    incremental_configs: ClassVar[list[ConfigEntry]]
-    configs: ClassVar[dict]
-    error: ClassVar[KafkaError]
+    restype: ConfigResource
+    restype_int: int
+    name: str
+    set_config_dict: dict[str, str]
+    incremental_configs: list[ConfigEntry]
+    configs: dict | None
```
Comment (kevinjacobs-delfi) [inline confluent_kafka-stubs/admin/_config.pyi:63]: corrected type

File: confluent_kafka-stubs/admin/_metadata.pyi [hunk 8]
```diff
@@ -4,61 +4,58 @@ This package is licensed under the Apache 2.0 License.
...
-    brokers: ClassVar[dict[int, "BrokerMetadata"] | None] = None
-    topics: ClassVar[dict[str, "TopicMetadata"] | None] = None
-    orig_broker_id: ClassVar[int] = -1
-    orig_broker_name: ClassVar[str] = ""
+    cluster_id: str
+    controller_id: int
+    brokers: dict[int, "BrokerMetadata"]
+    topics: dict[str, "TopicMetadata"]
+    orig_broker_id: int
+    orig_broker_name: str | None
```
Comment (kevinjacobs-delfi) [inline confluent_kafka-stubs/admin/_metadata.pyi:16]: corrected type

File: confluent_kafka-stubs/admin/_metadata.pyi [hunk 9]
```diff
@@ -4,61 +4,58 @@ This package is licensed under the Apache 2.0 License.
...
 class PartitionMetadata:
-    id: ClassVar[int]
-    leader: ClassVar[int]
-    replicas: ClassVar[list[int] | None] = None
-    isrs: ClassVar[list[int] | None] = None
-    error: ClassVar["KafkaError" | None] = None
+    id: int
+    leader: int
+    replicas: list[int]
+    isrs: list[int]
```
Comment (kevinjacobs-delfi) [inline confluent_kafka-stubs/admin/_metadata.pyi:38]: corrected types

File: confluent_kafka-stubs/admin/_metadata.pyi [hunk 10]
```diff
@@ -4,61 +4,58 @@ This package is licensed under the Apache 2.0 License.
...
-    protocol_type: ClassVar[Any | None] = None
-    protocol: ClassVar[Any] = None
-    members: ClassVar[list | None] = None
+    broker: "BrokerMetadata" | None
+    id: str | None
+    error: "KafkaError" | None
+    state: str | None
+    protocol_type: str | None
+    protocol: str | None
+    members: list[GroupMember] | None
```
Comment (kevinjacobs-delfi) [inline confluent_kafka-stubs/admin/_metadata.pyi:59]: corrected types

File: confluent_kafka-stubs/serializing_producer.pyi [hunk 11]
```diff
@@ -16,13 +16,13 @@ from .serialization import SerializationContext as SerializationContext
...
-        key: object | None = None,
-        value: object | None = None,
-        partition: int = -1,  # type: ignore[override]
+        value: str | bytes | None = None,
+        key: str | bytes | None = None,
+        partition: int = -1,
         on_delivery: Callable[[KafkaError | None, Message], None] | None = None,
         timestamp: float = 0,
-        headers: dict[str, bytes | None] | None = None,  # type: ignore[override]
+        headers: dict[str, str | bytes | None] | None = None,  # type: ignore[override]
```
Comment (kevinjacobs-delfi) [inline confluent_kafka-stubs/serializing_producer.pyi:27]: corrected type

File: confluent_kafka-stubs/serializing_producer.pyi [hunk 12]
```diff
@@ -16,13 +16,13 @@ from .serialization import SerializationContext as SerializationContext
...
     def __init__(self, conf: dict[str, Any]) -> None: ...
-    def produce(
+    def produce(  # type: ignore[override,reportIncompatibleMethodOverride]
         self,
         topic: str,
-        key: object | None = None,
-        value: object | None = None,
-        partition: int = -1,  # type: ignore[override]
+        value: str | bytes | None = None,
+        key: str | bytes | None = None,
```
Comment (kevinjacobs-delfi) [inline confluent_kafka-stubs/serializing_producer.pyi:23]: corrected types

**Comment by benbenbang** [comment]
@kevinjacobs-delfi 

please fix the conflicts first. Thanks

**Comment by kevinjacobs-delfi** [comment]
I don't see any remaining merge conflicts.  Can you show me where you are seeing them?

**Pull request closed without merging**